### PR TITLE
Introduced granulated Media Gallery ACL resources

### LINKED ISSUE #1487: Introduce granulated Media Gallery ACL resources and enforce for new media gallery
Story https://github.com/magento/adobe-stock-integration/issues/1485

### Description

Use `Magento_MediaGallery::media_gallery` ACL resource instead of `Magento_Cms::media_gallery`
The `Magento_MediaGallery::media_gallery` ACL resource should have the following subresources:
** Upload assets
** Delete assets
** Insert assets into the content
** Create folder
** Delete folder

The resources should be added to `app/code/Magento/MediaGallery/etc/acl.xml` file in magneto2 repo
The introduced ACL resources should be enforced for the new media gallery (Enhanced Media Gallery is enabled)
The ACL resources should affect if corresponding buttons/components are displayed to the user and if the controllers/services accept the actions

### PR DESCRIPTION
**Pull request by Nazar65**
### Related Pull Requests
https://github.com/magento/partners-magento2-infrastructure/pull/35

### Fixed Issues

Fixes magento/adobe-stock-integration#1487: Introduce granulated Media Gallery ACL resources and enforce for old media gallery

### Approval Request

https://jira.corp.magento.com/browse/MC-37384

### Manual testing scenarios (*)
```
** Upload assets
** Edit assets
** Delete assets
** Insert assets into the content
** Create folder
** Delete folder
```

1. Create new admin user
2. Try to apply all mentioned acl resources for this user one by time
3. Verify functionality in old media gallery && new media gallery

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/MediaGalleryUi/etc/acl.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        <resources>
+            <resource id="Magento_Backend::admin">
+                <resource id="Magento_Backend::content">
+                    <resource id="Magento_Backend::content_elements">
+                        <resource id="Magento_Cms::media_gallery">
+                            <resource id="Magento_Cms::upload_assets" title="Upload Assets" translate="title" sortOrder="70"/>
+                            <resource id="Magento_Cms::delete_assets" title="Delete Assets" translate="title" sortOrder="60"/>
+                            <resource id="Magento_Cms::insert_assets" title="Insert Assets into the content" translate="title" sortOrder="50"/>
+                            <resource id="Magento_Cms::create_folder" title="Create Folder" translate="title" sortOrder="40"/>
+                            <resource id="Magento_Cms::delete_folder" title="Delete Folder" translate="title" sortOrder="40"/>
```
Comment (BarnyShergold) [inline app/code/Magento/MediaGalleryUi/etc/acl.xml]: You've given all the other resource IDs different Sort Orders but Create & Delete have been given the same (40) - was this intentional? As it's not essential they are different this is not a blocker to review.

**Review by BarnyShergold — APPROVED** [review]

File: app/code/Magento/MediaGalleryUi/etc/acl.xml (hunk 1 above)
Comment (Nazar65) [reply app/code/Magento/MediaGalleryUi/etc/acl.xml]: thank you :+1:  , but this is still work in progress :) need to verify all cases

Comment (Nazar65) [reply app/code/Magento/MediaGalleryUi/etc/acl.xml]: @BarnyShergold could you please move this to the requested changes ?)

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Sort order for Create Folder & Delete Folder need to be different values

File: app/code/Magento/MediaGalleryUi/etc/acl.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * See COPYING.txt for license details.
+ */
+-->
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Acl/etc/acl.xsd">
+    <acl>
+        <resources>
+            <resource id="Magento_Backend::admin">
+                <resource id="Magento_Backend::content">
+                    <resource id="Magento_Backend::content_elements">
+                        <resource id="Magento_Cms::media_gallery">
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryUi/etc/acl.xml]: @Nazar65 let's move the media gallery ACL Resources to the proper namespace: `Magento_MediaGallery::<resource>` and move the `acl.xml` to the `Magento_MediaGallery` module

Comment (Nazar65) [reply app/code/Magento/MediaGalleryUi/etc/acl.xml]: ok

**Comment by Nazar65** [comment]
@magento run all test

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

File: app/code/Magento/MediaGalleryUi/composer.json [hunk 3]
```diff
@@ -12,7 +12,8 @@
         "magento/module-media-gallery-metadata-api": "*",
         "magento/module-media-gallery-synchronization-api": "*",
         "magento/module-media-content-api": "*",
-        "magento/module-cms": "*"
+        "magento/module-cms": "*",
+        "magento/module-media-gallery": "*"
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryUi/composer.json]: Probably it's better to move the ACL to API module to have the dependency on API module only

**Comment by sivaschenko** [comment]
@magento run all tests

**Comment by sivaschenko** [comment]
@magento run all tests

**Comment by sivaschenko** [comment]
@magento run all tests

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-8057](https://jira.corp.magento.com/browse/ENGCOM-8057) has been created to process this Pull Request

**Comment by sivaschenko** [comment]
@magento run all tests

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

Tested the following manual scenarios: 
- Upload assets
- Edit assets
- Delete assets
- Insert assets into the content
- Create folder
- Delete folder

Tested the following admin role resources in **New Media Gallery**:
Media Media Gallery:
- Insert assets into the content
- Upload assets
- Edit asset details
- Adobe Stock
- Save Preview Image
- License images
- Delete assets
- Create folder
- Delete folder

Tested creating a user role on Magento 2.4.0 + ASI 2.0.0-master, and upgrade instance to Magento 2.4.0 + ASI 2.1.0-master

Test scenarios which cover ACL functionality:
https://studio.cucumber.io/projects/131313/test-plan/folders/943908/scenarios/3218882
https://studio.cucumber.io/projects/131313/test-plan/folders/1051731/scenarios/3967955
https://studio.cucumber.io/projects/131313/test-plan/folders/1051731/scenarios/3968678

**Comment by sivaschenko** [comment]
@magento run all tests

**Comment by sivaschenko** [comment]
@magento run Functional Tests B2B,Sample Data Tests B2B

**Comment by sivaschenko** [comment]
SVC Approved. MFTF failures are caused by downtime of 3rd party service (InStorePickup modules). Ready for delivery

**Pull request merged**